command-expansion-server/Dockerfile

```diff
@@ -1,4 +1,9 @@
 FROM node:lts-alpine
+
 COPY . /app
 WORKDIR /app
+
+RUN npm ci
```

Comment:
I think it's best practice to not actually build the app when we build the Docker image. The Dockerfile should assume a build has already occurred so we can call `npm ci` in actual CI and cache the node_modules as needed.

Reply:
The problem I see with that is that the eventual execution environment is the docker container, NOT the CI environment. So for things like self-compiling binary packages, you can have a mismatch on target platform.

Reply:
Yes true. I think if we ever need to export binary packages we can resist. The general pattern in the repo is Gradle creates the build products in CI and our Dockerfiles copy them into place at Docker image build time. I don't want to diverge from that right now.

Reply:
It's not about us exporting binaries, but about packages we depend on. Any package we depend on that has a build process using node-gyp can cause us problems. Really any package that uses platform specific binaries. And because of node package resolution, it could be a dependency of a dependency.

Reply:
Ah yeah I hear you. For now we will just have to keep an eye on the system requirements in our dependencies and transitive dependencies and make sure our CI agents supports them.

Reply:
Just to elaborate @dyst5422 I don't think this is a bad idea. My main thing is I just want our build processes to be as consistent as possible across the different services in the repo so it's easier to maintain. So if we decide to add build steps to our Dockerfiles I think we should do it throughout the repo, and also spend some time to research a bit more on how to do it correctly so we can take advantage of things like agent dependency caching.

Reply:
I hear and agree with you. I have never done a workflow where build occurs in CI and then copied into containers, so apologies if I seem alarmist.